Cython/Compiler/ExprNodes.py

```diff
@@ -3082,25 +3099,39 @@ class CppIteratorNode(ExprNode):
     cpp_attribute_op = "."
     extra_dereference = ""
     is_temp = True
+    reversed = False
 
     subexprs = ['sequence']
 
+    def _get_begin_name(self):
+        """
+        Returns the name of function that returns the first iterator.
+        """
+        return "begin" if not self.reversed else "rbegin"
+
+    def _get_end_name(self):
+        """
+        Returns the name of function that returns the last iterator.
+        """
+        return "end" if not self.reversed else "rend"
+
```

Comment:
I would propose to combine `scope.lookup` into those functions and rename them to something like `lookup_sequence_begin`
They seem to be useless for anything else
And we can always split functions in the future

Reply:
Sure. Should I change the functions to something like this?
```python
def lookup_sequence_begin(self):
  begin_name = "begin" if not self.reversed else "rbegin"
  begin = self.sequence.type.scope.lookup(begin_name)
  return begin

def lookup_sequence_end(self):
  end_name = "end" if not self.reversed else "rend"
  end = self.sequence.type.scope.lookup(end_name)
  return end
```

Reply:
Personally I don't think so - the names are also used as strings in the code generation step.

Reply:
Thanks for the comments from both of you. I decided to keep my code unchanged unless there is a bug in my code or a better way to refactor it.

Reply:
We can `return begin_name, begin`, but it's not critical :]
Mostly I just didn't like the underscores in the function names

Reply:
I see. I was just trying to use underscores to indicate the functions were for internal use only. I don't think these functions are required externally. And the only thing I've come up with to make the code cleaner is to merge these functions into one like this:

```python
def _get_itr_func_names(self):
  """
  Returns the names of functions that returns the first and the last iterator.
  """
  return ("begin", "end") if not self.reversed else ("rbegin", "rend")
```

Reply:
Hmm, usually I also use underscores for implementation details, functions that should not be called outside the class/module
I am just not sure if those functoins have a use outside of this class

As I see, there is a call to get only `end_name`, so it's probably better to have two separate functions

Reply:
Maybe it's hard to tell at this stage. I'll leave it to someone who needs to modify it or comes up with a better implementation if possible.